c7n/resources/efs.py

```diff
@@ -39,7 +41,34 @@ class resource_type(TypeInfo):
         filter_type = 'scalar'
         universal_taggable = True
 
-    augment = universal_augment
+    def get_source(self, source_type):
+        return DescribeElasticFileSystem(self)
+
+
+class DescribeElasticFileSystem(DescribeSource):
+
+    def augment(self, resources):
+        client = local_session(self.manager.session_factory).client('efs')
+
+        with self.manager.executor_factory(
+            max_workers=min((10, len(resources) + 1))
```

Comment:
its probably worthwhile doing away with the thread pool, and just processing them one at a time, alternatively max concurrency 2-3.. very few apis support this number of ops a sec (s3 is an exception) without throttling.